src/timestamp.js

```diff
@@ -0,0 +1,62 @@
+/* @flow */
+
+export type Timestamp = Date
+
+export function InvalidTimestampError (obj: any) {
+  this.name = 'InvalidTimestampError'
+
+  this.message = `
+    Timestamps must be JavaScript UTC Date instances with second-only precision,
```

Comment:
Reviewing myself: a Date doesn't have to *be* UTC, it doesn't really make sense.